spec/u-namespace.md

```diff
@@ -0,0 +1,65 @@
+# MessageFormat 2.0 Unicode Namespace
+
+The `u:` namespace is reserved for use by the Unicode Consortium.
+
+## Options
+
+This section describes common **_<dfn>`u:` options</dfn>_** which each implementation SHOULD support
+for all _functions_ and _markup_.
+
+### `u:id`
+
+A string value that is included as an `id` or other suitable value
+in the formatted parts for the _placeholder_,
+or any other structured formatted results.
+
+Ignored when formatting a message to a string.
+
+Accepts string values, or values which can be stringified without error.
+For other values, a _Bad Option_ error is emitted
+and the `u:id` option is ignored.
+
+### `u:locale`
+
+A comma-delimited list of BCP 47 language tags,
+or an implementation-defined list of such tags.
+
+Replaces the _locale_ defined in the _function context_ for this _expression_.
+The value is ignored when set on _markup_.
+
+During processing, the `u:locale` option
+is always removed from the resolved mapping of _options_.
+
+Values matching the following ABNF are always accepted:
+```abnf
+u-locale-option = langtag *([s] "," [s] langtag)
+```
+using `langtag` as defined in [BCP 47](https://www.rfc-editor.org/rfc/bcp/bcp47.txt).
+Note that `langtag` is the rule for "normal language tags",
+and does not include private-use or grandfathered tags.
+
+Implementations MAY support additional language tags,
+such as private-use or grandfathered tags,
+or tags using `_` instead of `-` as a separator.
+When the value of `u:locale` is set by a _variable_,
+implementations MAY support non-string values otherwise representing locales.
+
+For unsupported values, a _Bad Option_ error is emitted
+and the value of the `u:locale` option is ignored.
+
+### `u:dir`
+
+Replaces the base directionality defined in
+the _function context_ for this _expression_.
+The value is ignored when set on _markup_.
```

Comment:
This is probably a mistake. Instead of ignoring it, we should say the behavior is undefined (an implementation of markup might make use of it?)

Reply:
I disagree; see my previous reply to this: https://github.com/unicode-org/message-format-wg/pull/846#discussion_r1721885800.

Reply:
In your previous comment, note that it doesn't prevent people from using it. This just says that it's not effective. My thinking is that we don't have a ton of implementation experience in this space. By not forbidding it (making it a Bad Option) we leave it in a kind of nether space ("valid" but ineffectual and tempting to abuse).

Reply:
I'd be fine emitting a Bad Option if this or `u:locale` is set on markup.

Reply:
Changed this to emit an error.